### ISSUE TITLE
Some problem about the human preference learning

### ISSUE THREAD

**Issue by Huwenbo-git** [HumanCompatibleAI/imitation#374]
I try to use the command "python -m imitation.scripts.train_preference_comparisons with pendulum" to train my agent. But I find that the performance of the algorithm is  terrible. The reward curve is not very good. The accuary of the reward function for the "cartpole" agent is very good, but the average cumulative return curve goes up and then suddenly goes down. For the cartpole agent, the average cumulative return always keeps on a constant. I think maybe some of the hyperparameter settings are wrong. Can you provide a good copy of the hyperparameter?

**Comment by AdamGleave** [comment]
Tagging @ejnnr who had some issues with Pendulum too, not sure we ever got to the bottom of that.

**Comment by ejnnr** [comment]
Actually, my issues were with Cartpole (and those haven't been resolved, @kmdanielduan told me that Cartpole didn't work for him either). Pendulum definitely worked at one point and I haven't tried it since then, I'll take a look later and see whether I can reproduce your bad results.

#371 contains hyperparameters that I think are usually a bit better than the ones in `master`, but they haven't been designed for Pendulum and still aren't very well tuned. I should also say that the preference comparisons implementation is far less mature than the other algorithms, so there might be some bug causing this (rather than just bad hyperparameters).

If you're ultimately interested in a different environment and only using Cartpole/Pendulum as test cases, @Huwenbo-git, then you could just try that one immediately. It worked pretty well for me without much hyperparameter tuning in several Mujoco environments, a grid world, and MountainCar (with a pretrained policy to get better exploration). Otherwise, I'm definitely planning to get to the bottom of these issues soon-ish. If you want to debug this yourself before then, I'd also be happy to give some pointers or answer any questions. And please do let me know about any other results you encounter or if you figure something out!

**Comment by ejnnr** [comment]
Pendulum seems to sometimes work and sometimes not, depending on the seed. Just did three runs and got final average returns of -216, -179 (which seem ok), and -1373 (which is not). Interestingly, the main difference in the learning curves seems to be about the policy, rather than the reward model. In particular, the bad run (pink in the plots below) gets a large entropy loss and high KL divergence (i.e. the policy changes a lot). In contrast, the reward model loss and accuracy are just as good as in the successful runs.

A good sanity check would be to log the reward according to the reward model (seems useful to do anyway). Might also be helpful to capture videos of the agent (I recall it learned really strange behaviour in Cartpole). Will do that when I continue debugging this issue.

![image](https://user-images.githubusercontent.com/6618882/140521535-6b29a4b5-0ba5-41aa-9545-a78e94dc2ff3.png)
![image](https://user-images.githubusercontent.com/6618882/140522024-fbbd92c2-c6ea-4bd9-a654-9b9aeab83d45.png)
![image](https://user-images.githubusercontent.com/6618882/140522081-b1a22037-f102-47c6-9d6e-67a152f2d2fc.png)

**Comment by AdamGleave** [comment]
@kmdanielduan I think you also ended up doing experiments in Pendulum and saw something similar?

**Comment by kmdanielduan** [comment]
I did similar experiments for CartPole, Pendulum and some self-designed MuJoCo environments. The algorithm seems to fail on CartPole (same as @ejnnr mentioned). It works for Pendulum for some seeds but not all. I used the default hyperparameters in `imitation.scripts.configs.train_preference_comparisons` except for setting PPO's entropy coefficient as 0.01, but haven't tried too much on the configurations in #371. 

<img width="896" alt="Screenshot 2021-11-09 at 14 30 55" src="https://user-images.githubusercontent.com/35738259/140873899-669fa0d3-ca22-4314-bbe2-c4713024fe94.png">
